federation/pkg/kubefed/init/init.go

```diff
@@ -586,6 +601,28 @@ func createControllerManager(clientset *client.Clientset, namespace, name, svcNa
 	return clientset.Extensions().Deployments(namespace).Create(dep)
 }
 
+func createFedConfigMap(config util.AdminConfig, fedName, kubeconfig, dnsZoneName string, dryRun bool) (*v1.ConfigMap, error) {
+	fedClientSet, err := config.FederationClientset(fedName, kubeconfig)
```

Comment:
I now think this construction should be pulled out to the caller method and should be reused here and and in `waitSrvHealthy()`